src/main/java/org/folio/dao/impl/FileDefinitionDaoImpl.java

```diff
@@ -0,0 +1,72 @@
+package org.folio.dao.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.sql.UpdateResult;
+import org.folio.dao.FileDefinitionDao;
+import org.folio.rest.jaxrs.model.FileDefinition;
+import org.folio.rest.persist.Criteria.Criteria;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.interfaces.Results;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import java.util.Optional;
+
+@Repository
+public class FileDefinitionDaoImpl implements FileDefinitionDao {
+  private final Logger logger = LoggerFactory.getLogger(FileDefinitionDaoImpl.class);
+  private static final String TABLE = "file_definitions";
+  private static final String ID_FIELD = "'id'";
+
+  private PostgresClientFactory pgClientFactory;
+
+  public FileDefinitionDaoImpl(@Autowired PostgresClientFactory pgClientFactory) {
+    this.pgClientFactory = pgClientFactory;
+  }
+
+  @Override
+  public Future<Optional<FileDefinition>> getById(String id, String tenantId) {
+    Promise<Results<FileDefinition>> promise = Promise.promise();
+    try {
+      Criteria idCrit = constructCriteria(ID_FIELD, id);
+       PgUtil.getById(TABLE, FileDefinition.class, id, okapiHeaders, vertxContext, GetDataExportFileDefinitionsByFileDefinitionIdResponse.class, asyncResultHandler);
+    } catch (Exception e) {
```

Comment:
all the try catch is also not required. the Criteria is not required. Pgutil takes care of it. 
Example:
https://github.com/folio-org/mod-finance-storage/blob/master/src/main/java/org/folio/rest/impl/BudgetAPI.java#L36

Reply:
same for the other places please. where save and update are being used